### PR TITLE
Fix nfs client

### PR DESCRIPTION
**Pull request by masayuki2009**
## Summary

- This PR consists of the following 3 commits
- commit1: Revert "fs: nfs: Do not bind to a local port in TCP mode"
- commit2: fs: nfs: Introduce CONFIG_NFS_DONT_BIND_TCP_SOCKET
- commit3: drivers: wireless: select NFS_DONT_BIND_TCP_SOCKET for GS2200M

## Impact

- None

## Testing

- Tested with spresense:wifi_smp
- Tested with sabre-6quad:netnsh_smp (QEMU)

### CODE REVIEW

**Comment by masayuki2009** [comment]
See https://github.com/apache/nuttx/pull/3707

**Comment by wengzhe** [comment]
Hi there, this patch works well with my NFS server on Ubuntu 22.04 (both with or without insecure option). Thanks a lot!

File: fs/nfs/Kconfig [hunk 1]
```diff
@@ -13,6 +13,15 @@ config NFS
 
 #if NFS
 
+config NFS_DONT_BIND_TCP_SOCKET
+	bool "Do not bind TCP socket"
```
Comment (xiaoxiang781216) [inline fs/nfs/Kconfig:17]: let's remove the prompt string to hide the option from defconfig

File: fs/nfs/Kconfig [hunk 2]
```diff
@@ -13,6 +13,15 @@ config NFS
 
 #if NFS
 
+config NFS_DONT_BIND_TCP_SOCKET
```
Comment (pkarashchenko) [inline fs/nfs/Kconfig:16]:
```suggestion
config NFS_NO_BIND_TCP_SOCKET
```
Not sure if this is better. I'm ok to keep existing name

File: fs/nfs/Kconfig [hunk 3]
```diff
@@ -13,6 +13,15 @@ config NFS
...
+	bool "Do not bind TCP socket"
+	default n
+	depends on NFS
```
Comment (pkarashchenko) [inline fs/nfs/Kconfig:19]: Do we need

```suggestion
	depends on TCP
```
?

File: fs/nfs/Kconfig (hunk 2 above)
Comment (masayuki2009) [reply fs/nfs/Kconfig:16]: @pkarashchenko 
Let me keep the name

File: fs/nfs/Kconfig (hunk 3 above)
Comment (masayuki2009) [reply fs/nfs/Kconfig:19]: @pkarashchenko 
If I applied your suggestion, the indentation would be incorrect.

image

So let me keep the config.

File: fs/nfs/Kconfig (hunk 1 above)
Comment (masayuki2009) [reply fs/nfs/Kconfig]: Done.

**Review by xiaoxiang781216 — APPROVED** [review]

File: fs/nfs/Kconfig (hunk 3 above)
Comment (pkarashchenko) [reply fs/nfs/Kconfig:19]: Very strange as if I add such a condition all looks fine on my end. Let me submit a PR, so you can try it out

Comment (pkarashchenko) [reply fs/nfs/Kconfig:19]: @masayuki2009 could you please check https://github.com/apache/nuttx/pull/9668?

File: fs/nfs/Kconfig [hunk 4]
```diff
@@ -13,6 +13,15 @@ config NFS
 
 #if NFS
 
+config NFS_DONT_BIND_TCP_SOCKET
+	bool
+	default n
+	depends on NFS
```
Comment (patacongo) [inline fs/nfs/Kconfig:19]: 19    depends on NFS

Within

    14 if NFS

Is redundant, is it not?

**Pull request merged**